beacon-chain/rpc/eth/helpers/sync.go

```diff
@@ -39,6 +60,9 @@ func ValidateSync(ctx context.Context, syncChecker sync.Checker, headFetcher blo
 }
 
 // IsOptimistic checks whether the latest block header of the passed in beacon state is the header of an optimistic block.
+// This is exposed to end-users who interpret `true` as "your Prysm beacon node is optimistically tracking head - your execution node isn't yet fully synced"
+// INFO: https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Node/getSyncingStatus for spec
+// INFO: https://github.com/prysmaticlabs/documentation/pull/429 for a PR to user-facing docs that introduces v1/node/syncing as a way to check beacon node + execution node sync status
```

Comment:
I don't think these comments are necessary. This is just a helper function - it's not exposed to end users, it's not implementing the spec, and I don't see the reason to include a PR here (happy to chat about it)

Even though I disagree with adding this here, it gave me an idea. We should probably add spec links to our grpc functions that implement the spec. :slightly_smiling_face:

Reply:
Thank you for this feedback @rkapka! Will incorporate this week.